src/sage/rings/function_field/drinfeld_modules/finite_drinfeld_module.py

```diff
@@ -512,12 +787,16 @@ def frobenius_norm(self):
             return self._frobenius_norm
         K = self.base_over_constants_field()
         n = K.degree(self._Fq)
-        char = self.characteristic()
+        r = self.rank()
+        p = self.characteristic()
         norm = K(self.coefficients()[-1]).norm()
-        self._frobenius_norm = ((-1)**n)*(char**(n/char.degree())) / norm
+        self._frobenius_norm = (-1) ** (n*r - n - r) \
+                               * norm**(-1) \
+                               * p ** (n//p.degree())
         return self._frobenius_norm
 
-    def frobenius_trace(self):
+    @cached_method
```

Comment:
Maybe caching in not needed here, since it's reimplemented in the method.

Reply:
Yep. I removed that on `frobenius_trace` and `frobenius_norm`.